### PR TITLE
PascalVOC seg fixes

### PR DESCRIPTION
**Pull request by lkdci**
PascalVOC seg fixes and changes:
* Support ignore label for better convergence and training.
* Correct train / val split with SBD extra annotation aka "pascal aug". as mentioned in deeplabv3 paper: 
```
We evaluate the proposed models on the PASCAL VOC 2012 semantic segmentation benchmark [20] which contains 20 foreground object classes and one background class.
The original dataset contains 1,464 (train), 1,449 (val), and 1,456 (test) pixel-level labeled images for training, validation, and testing, respectively.
The dataset is augmented by the extra annotations provided by [29], resulting in 10,582 (trainaug) training images
```

### CODE REVIEW

File: src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py [hunk 1]
```diff
@@ -96,3 +118,49 @@ def _get_pascal_labels(self):
...
+    validation set.
+    """
+    def __init__(self, **kwargs):
+        print(kwargs)
+        if any([kwargs.pop("list_file"), kwargs.pop("samples_sub_directory"), kwargs.pop("targets_sub_directory")]):
+            logger.warning("[list_file, samples_sub_directory, targets_sub_directory] arguments passed will not be used"
+                           " when passed to `PascalVOCAndAUGUnifiedDataset`. Those values are predefined for initiating"
+                           " the Pascal VOC + AUG training set.")
+        super().__init__(datasets=[
+            PascalVOC2012SegmentationDataSet(list_file="VOCdevkit/VOC2012/ImageSets/Segmentation/train.txt",
```
Comment (lkdci) [inline src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py:160]: Those arguments are hardcoded which is not a clean way to define those arguments. On the other hand this the proper way to use this class in the first place, and a different configuration will be semantically wrong (Not `PascalVOCAndAUGUnifiedDataset`). Similar approach can be found in `PascalVOCUnifiedDetectionTrainDataset` where `train_dataset_names = ["train2007", "val2007", "train2012", "val2012"]` are hardcoded.
If we do want the flexibility to change those parameters I suggest to pass dataset_params as follows:
```
train_dataset_params:
  voc_subset_params:
    list_file: 
    samples_sub_directory: 
    targets_sub_directory: 
  aug_subset_params:
    list_file: 
    samples_sub_directory: 
    targets_sub_directory: 
...
``` 
WDYT? @shaydeci @ofrimasad

**Review by ofrimasad — COMMENTED** [review]
a few minor comments

File: src/super_gradients/recipes/dataset_params/pascal_voc_segmentation_dataset_params.yaml [hunk 2]
```diff
@@ -1,55 +1,62 @@
...
+        fill_mask: 21
 
     - SegCropImageAndMask:
         crop_size: 512
         mode: center
 
 
 train_dataloader_params:
-  batch_size: 64
+  batch_size: 16
```
[the remark this answers is not in the document]
Comment (lkdci) [reply src/super_gradients/recipes/dataset_params/pascal_voc_segmentation_dataset_params.yaml:53]: yes, batch size is too high. it get OOM on a5000

File: src/super_gradients/recipes/dataset_params/pascal_voc_segmentation_dataset_params.yaml [hunk 3]
```diff
@@ -1,55 +1,62 @@
...
   num_workers: 8
-  drop_last: False
+  drop_last: True
   pin_memory: True
 
 val_dataloader_params:
-  batch_size: 200
+  batch_size: 16
```
[the remark this answers is not in the document]
Comment (lkdci) [reply src/super_gradients/recipes/dataset_params/pascal_voc_segmentation_dataset_params.yaml:59]: yes, batch size is too high. it get OOM on a5000

File: src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py [hunk 4]
```diff
@@ -1,14 +1,37 @@
...
+    "diningtable",
+    "dog",
+    "horse",
+    "motorbike",
+    "person",
+    "potted-plant",
+    "sheep",
+    "sofa",
+    "train",
+    "tv/monitor",
```
[the remark this answers is not in the document]
Comment (lkdci) [reply src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py:34]: ambigious is another name for ignore_index, and is not a class that should be predicted or calculate metrics on. Might cause confusion and errors when using this parameter for getting num_classes: `num_classes = len(dataset.classes)`. Besides it follows the same convention as in Cityscapes.

**Comment by lkdci** [comment]
@ofrimasad , will really appreciate your re-review see my comments above

File: src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py (hunk 1 above)
Comment (ofrimasad) [reply src/super_gradients/training/datasets/segmentation_datasets/pascal_voc_segmentation.py]: let's keep it hardcoded for now. I agree it is a unique case where these params are not expected to change
